### PR TITLE
1019 fix ci workflow

### PR DESCRIPTION
**Pull request by JimMoen**

### CODE REVIEW

**Comment by JimMoen** [comment]
kerl works fine in this PR's : [CI-build_slim_packages](https://github.com/emqx/emqx/actions/runs/3279868494/jobs/5399943867#step:4:207)

kerl reported github [Auth Failed](https://github.com/emqx/emqx/actions/runs/3280141977/jobs/5400603252#step:5:208) in Manually ran action.

**Review by id — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -18,8 +18,7 @@ jobs:
           path: source
           fetch-depth: 0
       - id: detect-profiles
-        working-directory: source
```
Comment (id) [inline .github/workflows/release.yaml:21]: not sure you can change this

Comment (id) [reply .github/workflows/release.yaml:21]: it might break the build

Comment (JimMoen) [reply .github/workflows/release.yaml:21]: `.github/actions/detect-profiles` doesn't accept `working-directory`.
[Prepare Run Failed](https://github.com/emqx/emqx/actions/runs/3265399016/workflow#L34)
[Prepare Run Succeeded](https://github.com/emqx/emqx/actions/runs/3280141977/workflow#L34)

**Comment by JimMoen** [comment]
Merge first and then continue to observe CI behavior

**Review by id — APPROVED** [review]

**Comment by JimMoen** [comment]
[Cache Hit](https://github.com/emqx/emqx/actions/runs/3311025917/jobs/5465964891#step:4:182) with key `otp-install-23.3.4.9-3-macos-11-static-ssl-disable-hipe-disable-jit`
but havn't hit with same key in [Previous Built](https://github.com/emqx/emqx/actions/runs/3279868494/jobs/5399943867#step:4:195).
So there were lots of cache build with the same key.
image

It seems a bug on `actions/cache@v2` I think. 🤔

After this #9174 merged:
- Future PRs will build the cache will be built in `build_slim_packages` workflow in PR actions running.
- And the cache will be used in the action `cross_build_packages` after they merged into `main-v4.*`.

**Review by id — APPROVED** [review]

**Review by zmstone — APPROVED** [review]

**Pull request merged**